### ISSUE TITLE
[BUG] TimeGapSplit requires unique index, but doesn't check for this

### ISSUE THREAD

**Issue by rubenvdg** [koaning/scikit-lego#193]
`TimeGapSplit` operates on the index of the `pandas.DataFrame` and implicitly assumes the indices are unique. If this is not the case, `TimeGapSplit` returns undesirable output, without raising an exception. For example:

```
import pandas as pd
import numpy as np
from datetime import timedelta

from sklego.model_selection import TimeGapSplit

df = (
    pd.DataFrame(
        data=np.random.randint(0, 30, size=(30, 4)),
        columns=list('ABCy'),
        index=np.ones(30)
    )
    .assign(date=pd.date_range(start='1/1/2018', end='1/30/2018')[::-1])
)
        
cv = TimeGapSplit(
    df=df,
    date_col='date',
    train_duration=timedelta(days=10),
    valid_duration=timedelta(days=2),
    gap_duration=timedelta(days=1)
)

X, y = df[['A', 'B', 'C']], df['y']

assert list(cv.split(X, y))[0][0][0] == slice(0, 30, None)
```

Possible remedies are:
- Requiring unique indices
- Copying the dataframe and resetting the index

**Comment by koaning** [comment]
Again, copy here feels safe enough.

**Comment by stephanecollot** [comment]
I could just add something like

`assert len(df.index.unique) == len(df)`

**Comment by stephanecollot** [comment]
Ok I'm fixing this

**Comment by stephanecollot** [comment]
This was fixed and merged in #227
@koaning can you close?
